oauth/authentication-oauth2.0.md

```diff
@@ -0,0 +1,245 @@
+|grant_type|`string`|REQUIRED. For this flow grant type should always equal to "client_credentials".|
+
+##### Response
+
+```json
+{
+    "access_token": "syPHeMY5H--kdRtfpoXTgYFF7LHgVOhIjOQ5QkIvSD68VZvc2_uAew.P07tEVThD5SqNCV_tFwbAg",
+    "token_type"  : "Bearer",
+    "expires_in"  : 9.8270549
+}
+```
+| Name | Type |Description |
+| ------------- |------------- | -------------|
+|access_token|`string`|Access token to use for subsequent API's.|
+|token_type|`string`|Will always be "Bearer".|
+|expires_in|`number`|Seconds in which the access token will expire, for the user application to handle.|
+
+##### Example
+
+```bash
+curl -X POST "https://app.solvexia.com/oauth/token" \
+     -H "Content-Type: application/json" \
+     -d '{"client_id":"DDF-AAFBD447-55432-475B-83DB-B5AD78878821","client_secret":"B5AD78878821","grant_type":"client_credentials"}'
+```
+
+## Authorize Code Flow
+
+Authorize Code Flow is for the client public applications.
+
+The flow contains the following steps:
+1. Redirect to SolveXia authorization endpoint
+2. SolveXia prompts user to login and consent
+3. Generate access token & refresh token by supplying authorization code and client id
+
+User can refresh the access token with refresh token
+
+#### 1. Redirect to SolveXia authorization endpoint
+
+```http
+GET
+https://[env].solvexia.com/authorize?client_id=your_client_id&response_type=code&redirect_uri=your_return_url
+```
+
+##### Parameters
+
+| Name | Type |Description |
+| ------------- |------------- | -------------|
+|client_id|`string`|REQUIRED. Client id that you received when you created an application in SolveXia.|
+|response_type|`string`|REQUIRED. will always be "code".|
+|redirect_uri|`string`|OPTIONAL. If not supplied it will use configured redirect url of the oauth application, if supplied it must match the oauth application redirect url.|
+|state|`string`|OPTIONAL. Specifies any string value that your application uses to maintain state between your authorization request and the authorization server's response.|
+
+After the SolveXia authorization server successfully verifies your parameters, it will redirect the user to the login page and, as a next step, to the authorization form.
+
+#### 2. SolveXia prompts user to login and consent
+![login](login.png)
+
+In this step, the user logs in and considers whether to grant your application the requested access. At this stage, SolveXia displays a consent form that shows the name of your application.
+The user can then authorize access to their SolveXia resources requested by your application or refuse the request.
+
+![authorize](authorize.png)
+
+If the user authorizes access, you will be redirected back to the supplied (or configured) redirect uri with the authorization code in the query string.
+If you supplied an additional state parameter in the first step, SolveXia will return that as well.
```

Comment:
Should be:
If you supplied an optional state parameter...

Reply:
Thanks, Bhavin. Fixed